crates/uv-distribution-filename/src/wheel.rs

```diff
@@ -326,6 +341,14 @@ impl Serialize for WheelFilename {
     }
 }
 
+impl CacheKey for WheelFilename {
+    fn cache_key(&self, state: &mut CacheKeyHasher) {
+        self.name.hash(state);
+        self.version.hash(state);
+        self.tags.hash(state);
```

Comment:
It seems these hashes are not stable across compiler releases, while `cache_key` is supposed to be.

We need to use `self.name.cache_key(state)`.
Which will require either implementing `CacheKey` for all types in `WheelFilename` or we can cut-off to a string using `format!("{}", self.tags).cache_key(state)`. 

I prefer cutting to string to avoid hash change if we restructure inner types, but no strong preference.


https://doc.rust-lang.org/std/hash/trait.Hash.html#portability

Reply:
I was hoping we could avoid the allocations but I guess you're right that this could have some negative side effects. Feel free to change.

Reply:
Cool, changing it. I think since we are going with `version-tags` and `version-digest`. I will also keep the digest to just be digest of the tags instead of full stem.